geth/node/manager.go

```diff
@@ -248,40 +188,31 @@ func (m *Manager) addPeer(url string) error {
 	return nil
 }
 
-func (m *Manager) removePeer(url string) error {
-	parsedNode, err := discover.ParseNode(url)
-	if err != nil {
-		return err
-	}
-	m.node.Server().RemovePeer(parsedNode)
-	return nil
-}
-
 // PeerCount returns the number of connected peers.
-func (m *Manager) PeerCount() int {
-	if !m.IsNodeRunning() {
+func (m *StatusNode) PeerCount() int {
+	if !m.IsRunning() {
 		return 0
 	}
 	return m.node.Server().PeerCount()
 }
 
-// NodeConfig exposes reference to running node's configuration
-func (m *Manager) NodeConfig() (*params.NodeConfig, error) {
+// Config exposes reference to running node's configuration
+func (m *StatusNode) Config() (*params.NodeConfig, error) {
 	m.mu.RLock()
 	defer m.mu.RUnlock()
 
-	if err := m.isNodeAvailable(); err != nil {
+	if err := m.isAvailable(); err != nil {
 		return nil, err
 	}
 	return m.config, nil
 }
 
 // LightEthereumService exposes reference to LES service running on top of the node
-func (m *Manager) LightEthereumService() (*les.LightEthereum, error) {
+func (m *StatusNode) LightEthereumService() (*les.LightEthereum, error) {
```

Comment:
Will you keep it or does it only stay in this first step?

Reply:
just this first step :)